elastic/kibana
# replace actionHash with uuid

Resolves: #149831

This PR, replaces the `actionHash` that is used as an identifier of the throttled actions in the task state, with `uuid`.

## To verify:
Run Es with `path.data`
Create a rule that is creating multiple alerts, in v8.7 (or main branch) with two actions, 
one action `For each alert` and `On custom interval` => `{throttle: 1h, summary: false, notifyWhen: onThrottleInterval}`
one action `Summary of alerts`  and `On custom interval`  => `{throttle: 1h, summary: true, notifyWhen: onThrottleInterval}`

Observe the task status like below:
```
{
  "alertTypeState": {...},
  "alertInstances": {
    "host-1": {
      ...
      "meta": {
        ...
        "lastScheduledActions": {
          ...
          "actions": {
            ".server-log:metrics.threshold.fired:1h": {
              "date": "2023-02-23T14:37:48.285Z"
            }
          }
        }
      }
    },
    "host-2": {
     ....
      "meta": {
       ...
        "lastScheduledActions": {
          ...
          "actions": {
            ".server-log:metrics.threshold.fired:1h": {
              "date": "2023-02-23T14:37:48.285Z"
            }
          }
        }
      }
    }
  },
  "summaryActions": {
    ".server-log:metrics.threshold.fired:1h": {
      "date": "2023-02-23T14:36:46.085Z"
    }
  },
}

```

Stop Es and Kibana, and switch to this branch.
Run ES with the same `path.data` again.
The rule still must be running as it was (throttling) and the action identifier in the task state should be replaced with uuid like shown below.

```
{
  "alertTypeState": {...},
  "alertInstances": {
    "host-1": {
      ...
      "meta": {
        ...
        "lastScheduledActions": {
          ...
          "actions": {
            "34f542f2-4bc1-4434-94d1-8db5a3a3d282": { <== here comes a uuid
              "date": "2023-02-23T14:37:48.285Z"
            }
          }
        }
      }
    },
    "host-2": {
     ....
      "meta": {
       ...
        "lastScheduledActions": {
          ...
          "actions": {
            "34f542f2-4bc1-4434-94d1-8db5a3a3d282": {
              "date": "2023-02-23T14:37:48.285Z"
            }
          }
        }
      }
    }
  },
  "summaryActions": {
    "34f542f2-4bc1-4434-94d1-8db5a3a3d282": { <=== and here
      "date": "2023-02-23T14:36:46.085Z"
    }
  },
}

```
